examples-gallery/plot_sliding_block.py

```diff
@@ -0,0 +1,306 @@
+
+# %%
+# plot the accuracy of the solution and the trajectories
+prob.plot_constraint_violations(solution)
+
+# %%
+prob.plot_trajectories(solution)
+
+# %% 
+# animate the solution
+if selektion in (0, 1):
+    duration = (num_nodes - 1) * solution[-1]
+times = np.linspace(0.0, duration, num=num_nodes)
+interval_value = duration / (num_nodes - 1)
+
+strasse1 = strasse(x, a, b)
+strasse_lam = sm.lambdify((x, a, b), strasse1, cse=True)    
+
+P0_x = solution[:num_nodes]
+P0_y = strasse_lam(P0_x, par_map[a], par_map[b])
+
+# find the force vector applied to the block
+alpha = sm.atan(strasse(x, a, b).diff(x))
+Pfeil = [F*sm.cos(alpha),  F*sm.sin(alpha)]
+Pfeil_lam = sm.lambdify((x, F, a, b), Pfeil, cse=True)     
+
+l1 = laenge * num_nodes
+l2 = (laenge + 1) * num_nodes
+Pfeil_x = Pfeil_lam(P0_x, solution[l1: l2], par_map[a], par_map[b])[0]
+Pfeil_y = Pfeil_lam(P0_x, solution[l1: l2], par_map[a], par_map[b])[1]
+
+# needed to give the picture the right size.
+xmin = np.min(P0_x)
+xmax = np.max(P0_x)
+ymin = np.min(P0_y)
+ymax = np.max(P0_y)
+
+
+def initialize_plot():
+    fig = plt.figure(figsize=(8, 8))
+    ax = fig.add_subplot(111)
+    ax.set_xlim(xmin-1, xmax + 1.)
+    ax.set_ylim(ymin-1, ymax + 1.)
+    ax.set_aspect('equal')
+    ax.set_xlabel('X-axis [m]')
+    ax.set_ylabel('Y-axis [m]')
+
+    if selektion == 0:
+        msg = f'speed was optimized'
+    elif selektion == 1:
+        msg = f'speed and energy optimized, weight of speed = {verstaerkung:.1e}'
+    else:
+        msg = f'energy optimized'
+
+    ax.grid()
+    strasse_x = np.linspace(xmin, xmax, 100)
+    ax.plot(strasse_x, strasse_lam(strasse_x, par_map[a], par_map[b]), 
+        color='black', linestyle='-', linewidth=1)  
+    ax.axvline(initial_state_constraints[x], color='r', linestyle='--', linewidth=1)                              
+    ax.axvline(final_state_constraints[x], color='green', linestyle='--', linewidth=1)                           
+
+# Initialize the block and the arrow
+    line1, = ax.plot([], [], color='blue', marker='o', markersize=12)                                              
+    pfeil   = ax.quiver([], [], [], [], color='green', scale=35, width=0.004)
+    return fig, ax, line1, pfeil, msg
+
+fig, ax, line1, pfeil, msg = initialize_plot()
+
+# Function to update the plot for each animation frame
+def update(frame):
+    message = (f'running time {times[frame]:.2f} sec \n' +  
+        f'the red line is the initial position, the green line is the final position \n' + 
+        f'the green arrow is the force acting on the block \n' +
+        f'{msg}' )
+    ax.set_title(message, fontsize=12)
+    
+    line1.set_data([P0_x[frame]], [P0_y[frame]])
+    pfeil.set_offsets([P0_x[frame], P0_y[frame]])                                                  
+    pfeil.set_UVC(Pfeil_x[frame], Pfeil_y[frame])
+    return line1, pfeil
+
+animation = FuncAnimation(fig, update, frames=range(len(P0_x)), 
+    interval=1000*interval_value, blit=True)
```

Comment:
Animation does not play when built.

Reply:
I do not know how to get animations in .py files, yet, only in ipynb files. Will need my son's help.
(so, I cannot test, yet)

Reply:
If you run `python examples-gallery/plot_sliding_block.py` the animation should play for you.